src/arrayMethodReduce.js

```diff
@@ -5,8 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+    let currentElementIndex = 0;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+      currentElementIndex = 1;
+    }
+
+    for (let i = currentElementIndex; i < this.length; i++) {
+      const currentElement = this[i];
```

Comment:
this variable is redundant

Reply:
it was added to better remember how it works, now it removed as useless